sklearn/datasets/_base.py

```diff
@@ -69,9 +69,14 @@ def clear_data_home(data_home=None):
 
 
 def _convert_data_dataframe(caller_name, data, target,
-                            feature_names, target_names):
+                            feature_names, target_names, sparse_data=False):
     pd = check_pandas_support('{} with as_frame=True'.format(caller_name))
-    data_df = pd.DataFrame(data, columns=feature_names)
+    if not sparse_data:
+        data_df = pd.DataFrame(data, columns=feature_names)
+    else:
+        data_df = pd.DataFrame.sparse.from_spmatrix(data,
+                                                    columns=feature_names)
```

Comment:
`from_spmatrix` is new in pandas=0.25.0. Our CI uses the latest pandas, our documentation says we need pandas>=0.18 for some examples. Should we bump that to 0.25? @NicolasHug @rth maybe?

Reply:
Pandas 0.18 was released in 2016 while 0.25.3 was released less than a year ago. We can update a bit but I think 0.25 is too recent.

I think it would be OK to raise an exception there for older versions of pandas saying that pandas 0.25+ is required for this functionality, and then skip the corresponding tests (and check that the exception is raised).

Reply:
we won't have a way to test for that _error_ in any way on our CI though, since we don't test for our minimum supported pandas.

Reply:
> we won't have a way to test for that error in any way on our CI though, since we don't test for our minimum supported pandas.

Fair enough, but we can still skip the test for pandas <0.25.

Reply:
yes, I think? Or odeally we'd actually have a CI which tests for the min supported pandas since we're doing more and more of it

Reply:
I'm more than happy to add an exception (or go with the none sparse version for the older pandas case).

But, I also think there should be a test with the oldest supported versions of all the dependencies, having one of those certainly helped astropy to notice issues. Happy to add such a job, but that's probably best to do outside of this PR (and I also need to familiarize myself with your CI approaches first).

Reply:
I am planning to update our CI and make sure we test our min dependencies this week.

(I want to update the CI before the global scikit-learn sprint.)